homeassistant/components/hyperion/light.py

```diff
@@ -362,6 +559,13 @@ def _update_client(self, json):
 
     async def async_added_to_hass(self):
         """Register callbacks when entity added to hass."""
+
+        self._dispatcher_instance_removed_unsub = async_dispatcher_connect(
+            self.hass,
+            SIGNAL_INSTANCE_REMOVED.format(self._unique_id),
+            self.async_remove,
+        )
```

Comment:
We can use `self.async_on_remove` to register the unsubscribe callbacks for removal on entity remove.
```suggestion
        self.async_on_remove(
            async_dispatcher_connect(
                self.hass,
                SIGNAL_INSTANCE_REMOVED.format(self._unique_id),
                self.async_remove,
            )
        )
```